### PR TITLE
Thread email notification & user notification settings

### PR DESCRIPTION
**Pull request by FlowFlorent**
### Description

Adds doc
- REST API: getInboxNotification, getRoomNotificationSettings, updateRoomNotificationSettings, deleteRoomNotificationSettings
- node: getInboxNotification, getRoomNotificationSettings, updateRoomNotificationSettings, deleteRoomNotificationSettings
- Webhook: ThreadEmailNotification

Adds to node package
- client: getInboxNotification, getRoomNotificationSettings, updateRoomNotificationSettings, deleteRoomNotificationSettings
- webhook: ThreadEmailNotification

### CODE REVIEW

File: packages/liveblocks-core/src/types/IWebSocket.ts [hunk 1]
```diff
@@ -48,12 +48,15 @@ export interface IWebSocket {
...
  *
+ *   10xx: client will reauthorize (just like 41xx)
  *   40xx: client will disconnect
  *   41xx: client will reauthorize
  *   42xx: client will retry without reauthorizing (currently not used)
  *
  */
 export enum WebsocketCloseCodes {
+  /** Normal close of connection, the connection fulfilled its purpose. */
+  CLOSE_NORMAL = 1000,
```
Comment (FlowFlorent) [inline packages/liveblocks-core/src/types/IWebSocket.ts:59]:
```suggestion
```

File: packages/liveblocks-core/src/types/IWebSocket.ts [hunk 2]
```diff
@@ -72,6 +75,8 @@ export enum WebsocketCloseCodes {
   MAX_NUMBER_OF_MESSAGES_PER_DAY_PER_APP = 4004,
   /** Room is full, disconnect */
   MAX_NUMBER_OF_CONCURRENT_CONNECTIONS_PER_ROOM = 4005,
+  /** The server kicked the connection from the room. */
+  KICKED = 4100,
```
Comment (FlowFlorent) [inline packages/liveblocks-core/src/types/IWebSocket.ts:79]:
```suggestion
```

File: packages/liveblocks-core/src/types/IWebSocket.ts [hunk 3]
```diff
@@ -48,12 +48,15 @@ export interface IWebSocket {
 /**
  * The following ranges will be respected by the client:
  *
+ *   10xx: client will reauthorize (just like 41xx)
```
Comment (FlowFlorent) [inline packages/liveblocks-core/src/types/IWebSocket.ts:51]:
```suggestion
```

**Review by marcbouchenoire — APPROVED** [review]
Left some comments about some JSDoc comments but otherwise LGTM 🚀

**Review by CTNicholas — APPROVED** [review]
Thanks @FlowFlorent! I've pushed my changes, approved.

**Pull request merged**